### ISSUE TITLE
Android Q Beta 5: Fails to access storage despite having permission

### ISSUE THREAD

**Issue by bughit** [TeamNewPipe/NewPipe#2470]
version: v0.16.2 from f-droid on Android Q Beta 5

There are two symptoms:

1.The audio download folder browser does not show any files/folders in `/storage/emulated/0`
2. Downloading audio to the default location (`/storage/emulated/0/Music/NewPipe`) fails with an error: `The file can not be created (code: 1000)`

The storage permission has been granted.

0.16.2 on Android 8.1 is fine.

**Comment by Stypox** [comment]
Could you please check if this issue exists in the current dev branch? (After #2149 was merged)
If you cannot compile the project by yourself and need an APK feel free to ask :-)

**Comment by kapodamy** [comment]
please provide a logcat

**Comment by bughit** [comment]
If you do a beta/preview through [releases](../releases) I can test that but I don't want to download apks from wherever.

**Comment by Stypox** [comment]
If you want I can guide you through the building process, but I don't think we will make a pre-release. You'll have to wait until the next full release.

**Comment by bughit** [comment]
> If you want I can guide you through the building process

It would make sense to add build directions to the repo. Then any bug reporter can decide if they are able or willing to follow them. 

`more-info-needed` would be appropriate if you tried and failed to repro this. Have you tried to repro, at least in the emulator?

**Comment by Stypox** [comment]
Well it is very simple to build, there are just 4 steps: download the project, open it in Android Studio, if a prompt comes up telling you to update gradle answer "No", click on build.

Sorry, I didn't even think about trying to reproduce this bug in an emulator... My pc is not powerful, so I never use emulators but prefer to debug directly on my phone. I will probably be able to create an emulator in 1/2 weeks, since currently my internet connection peaks at 30KiB/s and there are many GBs to download in order to get a full emulator working.
By the way `more-info-needed` means that this issue is on hold until new information comes up.

**Comment by Stypox** [comment]
Hopefully in a week 0.17.0 will be out, so you can test directly there. In #2483 you can find the preview download links for testing purposes.

**Comment by theScrabi** [comment]
You can test the RC versions already, over at the release v0.17.0 issue.

**Comment by bughit** [comment]
with NewPipe-v0.17.0_RC2

- The default download paths are old style (`/storage/emulated/0/...`), so downloads fail (could not create folder)
- browsing for download folder now works and creates a new style path (`content://com.android.externalstorage.documents/...`)
- browsing and selecting the folder is seemingly what grants access to it
- starting the download prompts for the storage permission, which is possibly no longer needed after new style browsing already granted access to the download folder
- audio download started but was consistently extremely slow ~32KB/s, multiple videos
  - 0.16.2 on 8.1 on the same wifi connection is megabytes/s
  - the slowness is specific to newpipe, chrome doing a fast.com speedtest maxes out the bandwidth

**Comment by gkeegan** [comment]
@bughit While it is strange that the speed decreased between NewPipe versions, it is likely that it will never reach the maximum speed shown on speed tests. No site ever really can achieve those speeds, even with an official app.

**Comment by bughit** [comment]
>  it is likely that it will never reach the maximum speed shown on speed tests

And who is asking for this? The point of the speedtest is to show that the device itself is not the limiting factor. The same content over the same connection is downloaded at 6MB/s by 0.16.2 on 8.1 and 30KB/s by 0.17rc2 on q beta5. So this is a newpipe issue (possibly caused by an android regression).

**Comment by gkeegan** [comment]
What are the speeds like on 0.16.2 with q beta 5?

**Comment by gkeegan** [comment]
@bughit could you post the speed issue in #2483 so it is sure to be resolved or recognized as an issue before 0.17.0 is released.

Also, could you try to see what the speeds are like on 0.16.2 with q beta5? This could help let people see if the speed issue is with recent changes to NewPipe or with Android itself.

**Comment by bughit** [comment]
> Also, could you try to see what the speeds are like on 0.16.2 with q beta5? 

I can try playing but it won't be an apples to apples comparison since 0.16.2 can't download and playing is likely a different code-path than downloading.

**Comment by kapodamy** [comment]
@bughit 
1_ This is for keep the paths selected by the user in 0.16.2, should be dropped and force the user migrate to SAF in another release or ¿maybe now?.
3_ same as the first point, compatibility reasons
5.0_ my mistake, audio download was single-thread, the true, it should be for subtitles.
5.2_ issued by a socket leak

**Comment by bughit** [comment]
By the way, vlc 3.1.7 with the classic storage permission, can browse storage folders and play files on q beta5

According to https://developer.android.com/preview/privacy/scoped-storage

> As of Android Q Beta 5, apps that target Android 9 (API level 28) or lower see no change, by default, to how storage works from previous Android versions.

So why does newpipe require changes?

>  This is for keep the paths selected by the user in 0.16.2

I am talking about new install defaults. They shouldn't be something that produces errors.

>  my mistake

Are you saying that that you know what's causing the slow audio download in 0.17? And will be fixing it?

**Comment by kapodamy** [comment]
_true the permission request is not required
_yes both things are done in #2486

**Comment by bughit** [comment]
Does anyone know the answer to the following?

> 
> By the way, vlc 3.1.7 with the classic storage permission, can browse storage folders and play files on q beta5
> 
> According to https://developer.android.com/preview/privacy/scoped-storage
> 
> > As of Android Q Beta 5, apps that target Android 9 (API level 28) or lower see no change, by default, to how storage works from previous Android versions.
> 
> So why does newpipe require changes?

It seems like 0.16.2 should have continued working on q.
